src/Assets.php

```diff
@@ -174,20 +173,37 @@ protected static function get_file_version( $file ) {
 	 * @param string $src       Full URL of the script, or path of the script relative to the WordPress root directory.
 	 * @param array  $deps      Optional. An array of registered script handles this script depends on. Default empty array.
 	 * @param bool   $has_i18n  Optional. Whether to add a script translation call to this file. Default 'true'.
+	 * @param bool   $enqueue   Optional. Whether the script should be enqueued instead of registered.
 	 */
-	protected static function register_script( $handle, $src, $deps = array(), $has_i18n = true ) {
+	protected static function register_script( $handle, $src, $deps = array(), $has_i18n = true, $enqueue = false ) {
 		$filename     = str_replace( plugins_url( '/', __DIR__ ), '', $src );
 		$ver          = self::get_file_version( $filename );
 		$deps_path    = dirname( __DIR__ ) . '/' . str_replace( '.js', '.deps.json', $filename );
 		$dependencies = file_exists( $deps_path ) ? json_decode( file_get_contents( $deps_path ) ) : array(); // phpcs:ignore WordPress.WP.AlternativeFunctions
 		$dependencies = array_merge( $dependencies, $deps );
 
-		wp_register_script( $handle, $src, $dependencies, $ver, true );
+		if ( $enqueue ) {
+			wp_enqueue_script( $handle, $src, $dependencies, $ver, true );
+		} else {
+			wp_register_script( $handle, $src, $dependencies, $ver, true );
+		}
```

Comment:
I think it will, but should verify that `wp_set_script_translations` still sets up the translations as expected when loaded late in the request.

Reply:
Good point! I tried creating a Spanish translation and it seems to work fine:
![image](https://user-images.githubusercontent.com/3616980/61536766-b947f480-aa35-11e9-99f7-25fd0813c628.png)